core/src/wallet/tezos/delegation/TezosLikeOriginatedAccount.cpp

```diff
@@ -62,9 +63,11 @@ namespace ledger {
             }
 
             static inline void update_balance(std::shared_ptr<api::Operation> const& op, BigInt& sum) {
-                auto value = BigInt(op->getAmount()->toString());
                 auto tzOp = op->asTezosLikeOperation();
                 auto tzTx = tzOp->getTransaction();
+                auto value = (tzTx->getType() == api::TezosOperationTag::OPERATION_TAG_TRANSACTION)
```

Comment:
Maybe we could just not call `update_balance` when that was not a `OPERATION_TAG_TRANSACTION`?
Since we know this will not change the sum anyway

Reply:
update_balance is called by a generic code (shared between several coins), we cannot use tezos specific functions (as getType)

Reply:
I see. Then maybe quit the function asap?
```
if (tzTx->getType() != api::TezosOperationTag::OPERATION_TAG_TRANSACTION)
    return;
```

Reply:
sum can be impacted by fees (even if value is equal to 0). We cannot quit the function